src/main/java/dao/PieceDao.java

```diff
@@ -0,0 +1,129 @@
+package dao;
+
+import domain.position.Position;
+import domain.unit.Team;
+import domain.unit.UnitType;
+import entity.Piece;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PieceDao {
+    private static final String SERVER = "localhost:13306"; // MySQL 서버 주소
+    private static final String DATABASE = "janggi"; // MySQL DATABASE 이름
+    private static final String OPTION = "?useSSL=false&allowPublicKeyRetrieval=true&serverTimezone=UTC";
+    private static final String USERNAME = "root"; //  MySQL 서버 아이디
+    private static final String PASSWORD = "root"; // MySQL 서버 비밀번호
+
+    public Connection getConnection() {
+        try {
+            return DriverManager.getConnection("jdbc:mysql://" + SERVER + "/" + DATABASE + OPTION, USERNAME, PASSWORD);
+        } catch (final SQLException e) {
+            System.err.println("DB 연결 오류:" + e.getMessage());
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public void save(final Piece piece, final String roomId) {
+        final var query = """
+                INSERT INTO piece(position_x, position_y, piece_type, team, room_id)
+                VALUES (?, ?, ?, ?, ?)
+                """;
+        try (final var connection = getConnection();
+             final var ppst = connection.prepareStatement(query)) {
+            ppst.setInt(1, piece.positionX());
+            ppst.setInt(2, piece.positionY());
+            ppst.setString(3, piece.unitType().toString());
+            ppst.setString(4, piece.team().toString());
+            ppst.setString(5, roomId);
+            ppst.executeUpdate();
+        } catch (final SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public List<Piece> findBoardsByRoomId(String roomId) {
+        final var query = """
+                SELECT * FROM piece
+                WHERE room_id = ?
+                """;
+        final List<Piece> piece = new ArrayList<>();
+        try (final var connection = getConnection();
+             final var ppst = connection.prepareStatement(query)) {
+            ppst.setString(1, roomId);
+            final var resultSet = ppst.executeQuery();
+            while (resultSet.next()) {
+                piece.add(new Piece(
+                        resultSet.getLong("piece_id"),
+                        resultSet.getInt("position_x"),
+                        resultSet.getInt("position_y"),
+                        UnitType.valueOf(resultSet.getString("piece_type")),
+                        Team.valueOf(resultSet.getString("team"))
+                ));
+            }
+        } catch (final SQLException e) {
+            throw new RuntimeException(e);
+        }
+        return piece;
```

Comment:
사소할 수 있지만, try 내부에서 return을 해도 되지 않을까요? 👀 
```suggestion
            }
            return piece;
        } catch (final SQLException e) {
            throw new RuntimeException(e);
        }
```